File: docs-web.resources/com.ibm.docs.web.resources/WebContent/js/wseditor/js/listener/Listener.js
/* ***************************************************************** */
/*                                                                   */
/* IBM Confidential                                                  */
/*                                                                   */
/* IBM Docs Source Materials                                         */
/*                                                                   */
/* (c) Copyright IBM Corporation 2012. All Rights Reserved.          */
/*                                                                   */
/* U.S. Government Users Restricted Rights: Use, duplication or      */
/* disclosure restricted by GSA ADP Schedule Contract with IBM Corp. */
/*                                                                   */
/* ***************************************************************** */

dojo.provide("websheet.listener.Listener");
dojo.declare("websheet.listener.Listener",null,{
	/*boolean*/preCondition:function(event)
	{
		// to be implemented by each listener instance
		return true;
	},
	notify:function(caster, event)
	{
		//implement by each listener instance
	},
	startListening:function(caster)
	{
		caster.addListener(this);
	},
	endListening:function(caster)
	{
		caster.removeListener(this);
	},
	isListening:function(caster)
	{
		return caster.hasListener(this);
	}
});

File: docsapp/com.ibm.concord.spreadsheet/src/com/ibm/concord/spreadsheet/document/model/Listener.java
package com.ibm.concord.spreadsheet.document.model;

public abstract class Listener implements IListener
{
//  public boolean needListen(NotifyEvent e)
//  {
//    return true;
//  }
  
  public abstract void notify(Broadcaster caster, NotifyEvent e);
  
  public void startListening(Broadcaster caster)
  {
    caster.addListener(this);
  }
  
  public void endListening(Broadcaster caster)
  {
    caster.removeListener(this);
  }
}
